File: py3_solution/3_LSWRC.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        cursub = ''
        maxlen, curlen = 0, 0
        for val in s:
            if val not in cursub:
                cursub = ''.join((cursub, val))
            else:
                curlen = len(cursub)
                if maxlen < curlen:
                    maxlen = curlen
                cursub = cursub.split(val)[-1]
                cursub = ''.join((cursub, val))
        curlen = len(cursub)
        if maxlen < curlen:
            maxlen = curlen

        return maxlen

File: cpp_solution/3_LSWRC.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        map<char, int> m;
        int res = 0;
        int lastRepPose = -1;
        
        for(int i = 0; i < s.size(); i++) {
            if(m.find(s[i]) != m.end() && lastRepPose < m[s[i]]) {
                lastRepPose = m[s[i]];
            }
            if (i - lastRepPose > res) {
                res = i - lastRepPose;
            }
            m[s[i]] = i;
        }
        
        return res;
    }
};
